extensions/xslt/runtime/src/main/java/org/apache/camel/quarkus/component/xslt/CamelXsltRecorder.java

```diff
@@ -46,8 +56,7 @@ protected void configureEndpoint(
                 String remaining,
                 Map<String, Object> parameters) throws Exception {
 
-            final String fileName = FileUtil.stripExt(remaining, true);
-            final String className = StringHelper.capitalize(fileName, true);
+            final String className = uriResolver.getTransletClassName(remaining);
```

Comment:
can you start working on the upstream component to support this feature natively ?

Reply:
Not sure how that would make sense there because the upstream has no notion of build time/ runtime distinction? I mean a getTransletClassName() method could possibly be added to the upstream's XsltUriResolver, but the RuntimeUriResolver proposed here is quite different from it and am rather skeptical about letting RuntimeUriResolver to extend  XsltUriResolver. Or you mean something else?

Reply:
I mean it would be nice to check what we need to to upstream to avoid having to subclass the component here.

Reply:
OK, created https://github.com/apache/camel-quarkus/issues/441